gobblin-core/src/main/java/gobblin/source/extractor/filebased/FileBasedExtractor.java

```diff
@@ -41,32 +40,32 @@
  *            type of schema
  * @param <D>
  *            type of data record
- * @param <K>
- *            key type of the command output
- * @param <V>
- *            value type of the command output
  */
-public class FileBasedExtractor<S, D> implements Extractor<S, D> {
-  private Logger log = LoggerFactory.getLogger(FileBasedExtractor.class);
+public abstract class FileBasedExtractor<S, D> implements Extractor<S, D> {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(FileBasedExtractor.class);
+
+  protected final WorkUnit workUnit;
+  protected final WorkUnitState workUnitState;
+  protected final FileBasedHelper fsHelper;
+  protected final List<String> filesToPull;
+
+  protected final Closer closer = Closer.create();
```

Comment:
Can closer objects be re-used? e.g. can you call closer.register(), closer.close(), and then closer.register() again? If not then we will have to create a new closer instance for each file downloaded.